lib/mayaUsd/render/vp2RenderDelegate/mesh.h

```diff
@@ -84,6 +84,9 @@ struct HdVP2MeshSharedData
 #ifdef HDVP2_ENABLE_GPU_COMPUTE
     MSharedPtr<MeshViewportCompute> _viewportCompute;
 #endif
+
+    //! Fallback color changed
+    bool _fallbackColorDirty { false };
```

Comment:
One way would be to use dirty bits - add a custom one here https://github.com/Autodesk/maya-usd/blob/600c31abcd68ef3fe0047b28fb5f33f6edd0872f/lib/mayaUsd/render/vp2RenderDelegate/mesh.h#L171

Then you don't have to deal with clearing it that much since they are all cleared at the end of the repr update.